enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/AbstractBanDependencies.java

```diff
@@ -191,4 +207,19 @@ public void setSearchTransitive( boolean theSearchTransitive )
         this.searchTransitive = theSearchTransitive;
     }
 
+    /**
+     * @return the reactorProjects
+     */
+    public List<MavenProject> getReactorProjects()
+    {
+        return reactorProjects;
+    }
+
+    /**
+     * @param reactorProjects the reactorProjects to set
+     */
+    public void setReactorProjects( List<MavenProject> reactorProjects )
+    {
+        this.reactorProjects = reactorProjects;
```

Comment:
make a copy to avoid exposing mutable internal state.

Reply:
```suggestion
        this.reactorProjects = java.util.Collections.unmodifiableList( reactorProjects );
```

Reply:
Rebased the PR on latest master, and applied changes as @svenlange suggested :)

Reply:
@svenlange - but this will only be unmodifiable if referenced via `this.reactorProjects`. External code can still alter original `reactorProjects` that was sent to `setReactorProjects` method...

Reply:
As it's long time ago when I filed this PR, I can not recall why the getter and setter of `reactorProjects` is required. Local build looks OK without the getter and setter . :(

Reply:
@elharo @svenlange @pzygielo 
I've tested and confirmed that the getter and setter of `reactorProjects` are redundant. They are removed in latest commit.

Reply:
Great thanks!

Reply:
But this PR is still stuck cause @khmarbaise is not responding. Can anyone else review and decide, @elharo ?

Reply:
@elharo Friendly ping :)